app/portainer/views/account/CreateAccessToken/CreateAccessToken.tsx

```diff
@@ -0,0 +1,107 @@
+import { PropsWithChildren, useState } from 'react';
+
+import { FormControl } from '@/portainer/components/form-components/FormControl';
+import { TextInput } from '@/portainer/components/form-components/Input';
+import { Button } from '@/portainer/components/Button';
+import { Heading } from '@/portainer/components/form-components/Heading';
+import { TextTip } from '@/portainer/components/Tip/TextTip';
+import { Code } from '@/portainer/components/Code';
+import { CopyButton } from '@/portainer/components/Button/CopyButton';
+
+import styles from './CreateAccessToken.module.css';
+
+interface AccessTokenResponse {
+  rawAPIKey: string;
+}
+
+export interface Props {
+  // userId for whom the access token is generated for
+  userId: number;
+
+  // onSubmit dispatches a successful matomo analytics event
+  onSubmit: (description: string) => Promise<AccessTokenResponse>;
+
+  // onSuccess is called when upon clicking the done button
+  onSuccess: () => void;
+
+  // onError is called when an error occurs; this is a callback to Notifications.error
+  onError: (heading: string, err: unknown, message: string) => void;
+}
+
+export function CreateAccessToken({
+  onSubmit,
+  onSuccess,
+  onError,
+}: PropsWithChildren<Props>) {
+  const [description, setDescription] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [accessToken, setAccessToken] = useState('');
+
+  // TODO: use axios or API service
+  async function generateAccessToken() {
+    if (isLoading) {
+      return;
+    }
+
+    setIsLoading(true);
+    try {
+      const response = await onSubmit(description);
+      setAccessToken(response.rawAPIKey);
+    } catch (err) {
+      onError('Failure', err, 'Failed to generate access token');
+    } finally {
+      setIsLoading(false);
+    }
+  }
+
+  function getError() {
+    if (description.length === 0) {
+      return 'this field is required';
+    }
+    if (description.replaceAll(' ', '') !== description) {
+      return 'this field cannot contain spaces';
+    }
+    return '';
+  }
+
+  return (
+    <>
+      <div>
+        <FormControl inputId="input" label="Description" errors={getError()}>
+          <TextInput
+            id="input"
+            onChange={(value) => setDescription(value)}
+            type="text"
+            value={description}
+          />
+        </FormControl>
+        <Button
+          disabled={!!getError()}
+          onClick={generateAccessToken}
+          className={styles.addButton}
+        >
+          Add access token
+        </Button>
+      </div>
+      {accessToken && (
+        <>
+          <div className="col-sm-12 form-section-title">
```

Comment:
isn't Heading with the class `form-section-title`?

Reply:
want a horizontal line after the component (which this class seems to achieve)

Reply:
yes, but then why not
```
<FormSectionTitle>
   <FormSectionTile>...</>
   ...
</>
```
or just two components side by side.
since FormSectionTitle just uses this class, we can use the component instead of the class.

Reply:
IMO, just create a hr component, or style the hr tag

Reply:
IMO, just create a hr component, or style the hr tag

this is from ssl-cert in settings (using hr):

![image](https://user-images.githubusercontent.com/1381655/143189797-3d2416df-78c1-408e-8bab-c810ec5f5486.png)

I know it's grey and not black, but you can add a class to color the border in black

Reply:
to conclude, use hr, because a duplicate `form-section-title` isn't obvious

Reply:
agreed, its not obvious/explicit. Updated accordingly 👍